docs/architecture/runtime-agent-contract.md

```diff
@@ -0,0 +1,27 @@
+# Runtime Agent contract
+
+## Overview
+
+This document describes contract between [Management Plane](../terminology.md#management-plane) and [Runtime Agent](../terminology.md#runtime-agent). Main responsibilities of the Runtime Agent are:
+- [Establishing trusted connection](#establishing-trusted-connection)
+- [Renewing trusted connection](#renewing-trusted-connection)
+- [Configuring the runtime](#configuring-the-runtime)
+
+## Establishing trusted connection
+
+Runtime Agent automates the process of [establishing trusted connection](./establishing-trusted-connection.md#establishing-trusted-connection) between [Runtime](../terminology.md#runtime) and Managemt Plane. To initialy trigger the process the [Runtime Provisioner](../terminology.md#mp-runtime-provisioner) creates the instance of CustomResource that holds the one-time token to enable the pairing. Runtime Agent triggered by the creation of CustomResource performs the pairing process and in the end the trusted connection between the Runtime and the Management Plane is established. At this point further communication between Runtime Agent and Management Plane, and [Application](../terminology.md#application) and Runtime is possible.
```

Comment:
That description explains the flow and then implementation details for Kyma as a Runtime. That's good. But we need also genberal contract. That means what information will be included in the token etc. 
For example, we will need information about tenant, scenario etc.

Reply:
@PK85 Why do we need information like tenant and labeling (eg.scenario) on the runtime agent? I assume that runtime agent need only the Runtime ID. Using runtime ID agent fetches the runtime configuration from Management Plane.